lib/common/output_text.c

```diff
@@ -335,7 +335,9 @@ pcmk__formatted_vprintf(pcmk__output_t *out, const char *format, va_list args) {
 
     CRM_ASSERT(out != NULL);
 
-    len = vfprintf(out->dest, format, args);
+    if (pcmk__str_eq(out->fmt_name, "text", pcmk__str_none)) {
```

Comment:
I did this for now as a quick fix. Ultimately we might want to add a manual print option to `pcmk__output_t`, so that we can call a member function in default formatters, instead of calling a function that's defined in `output_text.c` for text output objects.

If we don't give every output type its own manual print function, then we should probably move the functionality to a common file and guard it by output type like this.

Reply:
Adding a new `->print` function seems like a reasonable approach.  I can work on that really quick, unless you've already got something halfway done

Reply:
Looking at this again, there's very few places where this function is actually used in a default formatter.  It was really intended to be used in building text message functions and that's it.  I'd say that any other places it's being used is a bug and should be corrected there instead.  A manual printing function doesn't make sense for formatters other than text - what does printing to the screen look like for XML, or logs, or curses?

Aside from fixing those other usages, perhaps what we should do in pcmk__formatted_printf is return if fmt_name is "none", do the print if it's "text", and assert otherwise.

Note that everything here also applies to pcmk__indented_(v)printf.

Reply:
See #2848.